spec/requests/providers_spec.rb

```diff
@@ -596,6 +631,35 @@ def token(connection)
           )
           expect(provider.authentication_token(:hawkular)).to eq(token(hawkular_connection))
         end
+
+        it "supports provider with multiple endpoints creation and prometheus" do
+          api_basic_authorize collection_action_identifier(:providers, :create)
+
+          post(api_providers_url, :params => gen_request(:create, sample_containers_multi_end_point_with_prometheus))
+
+          expect(response).to have_http_status(:ok)
+          expected = {"id"   => a_kind_of(String),
+                      "type" => containers_class,
+                      "name" => "sample containers provider with multiple endpoints and prometheus"}
+
+          results = response.parsed_body["results"]
+          expect(results.first).to include(expected)
+
+          provider_id = results.first["id"]
+          provider = ExtManagementSystem.find(provider_id)
+          expect(provider).to have_endpoint_attributes(default_connection["endpoint"])
+          expect(provider.authentication_token).to eq(token(default_connection))
+
+          expect(provider.connection_configurations.prometheus.endpoint).to have_endpoint_attributes(
+            prometheus_connection["endpoint"]
+          )
+
+          expect(provider.connection_configurations.prometheus_alerts.endpoint).to have_endpoint_attributes(
+            prometheus_alerts_connection["endpoint"]
+          )
+          expect(provider.authentication_token(:prometheus)).to eq(token(prometheus_connection))
+          expect(provider.authentication_token(:prometheus_alerts)).to eq(token(prometheus_alerts_connection))
```

Comment:
I'm not sure this tested what we wanted to test.
AFAICT it only checks parent endpoints got created correctly; we mainly want to test child manager got created (and can see the endpoint it needs).

Consider also adding test for deletion of prometheus endpoint and maybe of whole provider resulting in child manager deletion.

Reply:
@cben testing the child manager got created should probably not be done in the API tests. If I understand correctly, API tests are supposed to test the interface, not the implementation, and the child manager is definitely an implementation detail.

Reply:
I was thinking of `it doesnt blow up` type tests

Should the sub-manager logic go away(which i feel it might) then we will need to change these spec tests so endpoint creation is enough IMO.

Reply:
> API tests are supposed to test the interface, not the implementation, and the child manager is definitely an implementation detail.

Well, I really want an integration test that implementation reacted correctly to API *somewhere*. 
 Do you see a better place/way to test it?
OK, in principle, if we have a test in our repo that directly creating parent endpoint creates the child (do we?), and test that API creates endpoints, then *in theory* we're covered.  I have low trust for theory in this case though :-)

> I was thinking of it doesnt blow up type tests

Hmm, yes, for this specific failure it was blowing up, that's enough.